src/libEth/Eth.cpp

```diff
@@ -73,28 +73,42 @@ Json::Value populateReceiptHelper(
   ret["r"] = GetR(sig);
   ret["s"] = GetS(sig);
 
+  switch (tx.GetVersionIdentifier()) {
+    case TRANSACTION_VERSION:
+      ret["type"] = "0x0"; // FIXME?
```

Comment:
I wasn't sure if it made sense to return `"type": "0x0"` for Zilliqa transactions when you call `eth_getTransactionByHash` on them. It is the current behaviour, but I figured it might make sense to use the type to differentiate non-Eth transactions too?

We'd want to pick a value which is sufficiently unlikely to ever be used by an Ethereum transaction type.

Do you think this is worth changing now or should I just stick with the existing behaviour of returning `"type": "0x0"` for both Ethereum legacy transactions and Zilliqa transactions?

Reply:
OTOH maybe if we return a `type` outside of `{0,1,2}`, clients like Ethers will just explode when trying to parse such transactions...

Reply:
I had a look and other Ethereum-compatible chains are using other `type`s to represent their own things (e.g. [Optimism](https://github.com/ethereum-optimism/optimism/blob/develop/specs/deposits.md#the-deposited-transaction-type) uses a `type` of `0x7E` for its own transactions).

Reply:
Those ^^^ are good things to write in a comment, so when someone sees // FIXME, they don't have to search github to work out what you mean :-)